OpenSim/Simulation/SimbodyEngine/Joint.cpp

```diff
@@ -640,6 +629,26 @@ void Joint::updateFromXMLNode(SimTK::Xml::Element& aNode, int versionNumber)
                 childNameElt->setValue(childFrameName + "_offset");
             }
         }
+
+        // Version 30506 replaced Joint's CoordinateSet with a "coordinates"
+        // list property.
+        if (documentVersion <= 30506) {
+            if (aNode.hasElement("CoordinateSet")) {
+                auto coordSetIter = aNode.element_begin("CoordinateSet");
+                auto coordIter = coordSetIter->getRequiredElement("objects")
```

Comment:
I am pretty sure `objects` can be empty. A `WeldJoint` does have a `CoordinateSet` that is empty. There is also nothing stopping users from deleting all Coordinates in a `CoordinateSet` and in that case it should not attempt to add Coordinates to the property list. I think the check that `coordIter != objects_node->element_end()` is prudent.